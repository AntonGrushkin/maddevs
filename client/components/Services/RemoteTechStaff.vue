<template>
  <section id="remote-tech-staff" class="remote-tech-staff main-parent-item">
    <div class="container">
      <div class="remote-tech-staff__content-wrapper">
        <h2 class="remote-tech-staff__main-title main-title">Remote Tech Staff</h2>
        <div class="remote-tech-staff__flex-wrapper flex-wrapper">
          <RemoteTechStaffContent
            :title="teams.title"
            :subTitle="teams.subTitle"
            :paragraph="teams.paragraph"
            :buttonInnerText="teams.buttonInnerText"
            :modalWindowName="teams.modalWindowName"
            class="teams"
          />
          <RemoteTechStaffContent
            :title="individuals.title"
            :subTitle="individuals.subTitle"
            :paragraph="individuals.paragraph"
            :buttonInnerText="individuals.buttonInnerText"
            :modalWindowName="individuals.modalWindowName"
            class="individuals"
          />
        </div>
      </div>
    </div>
  </section>
</template>

<script>
import RemoteTechStaffContent from '@/components/Services/RemoteTechStaffContent';

export default {
  name: 'RemoteTechStaff',
  components: {
    RemoteTechStaffContent
  },
  data() {
    return {
      teams: {
        title: 'Teams',
        subTitle: 'Tired of 20 or 30 people not delivering you any features?',
        paragraph: 'How about only dealing with smart heads and capable hands? Enjoy our proactive rockets who will ask you to get them more work.',
        buttonInnerText: 'Get a team of ultra-fast coders',
        modalWindowName: 'teams-modal'
      },
      individuals: {
        title: 'Individuals',
        subTitle: 'Seeing one, two, five guys working on a minor thing? Ha!',
        paragraph: 'We are NOT going to bring you more people than you actually need. We\'ll just send one or two of our rangers to your frontier, and you\'ll enjoy peace of mind about IT.',
        buttonInnerText: 'Get individual proactive rockets',
        modalWindowName: 'individuals-modal'
      }
    };
  }
};
</script>

<style lang="scss" scoped>
  @import '../../assets/styles/vars';
  @import '../../assets/styles/get-vw';

  .remote-tech-staff {
    padding: get-vw(211px) 0 get-vw(105px);
    background-color: $bgcolor--black;

    &__main-title {
      padding-bottom: get-vw(65px);
      color: $text-color--red;
    }
  }

  @media only screen and (min-width: 1024px) {
    .remote-tech-staff {
      /deep/.teams {
        .remote-tech-staff__sub-title.sub-title {
          margin-right: get-vw(10px);
        }
      }
    }
  }

  @media only screen and (max-width: 878px) {
    .remote-tech-staff {
      &__flex-wrapper {
        flex-direction: column;
      }

      &__outsourcing-content {
        max-width: 100%;
      }
    }

    .individuals {
      padding-top: get-vw(104px, 878);
    }
  }

  @media only screen and (max-width: 768px) {
    .remote-tech-staff {
      padding: get-vw(140px, 768) 0 get-vw(70px, 768);

      &__main-title {
        padding-bottom: get-vw(39px, 768)
      }
    }
  }

  @media only screen and (max-width: 576px) {
    .remote-tech-staff {
      padding: get-vw(120px, 576) 0 get-vw(105px, 576);
    }
  }

  @media only screen and (max-width: 520px) {
    .remote-tech-staff {
      padding: get-vw(144px, 520) 0 23vw 0;
    }
  }
</style>
