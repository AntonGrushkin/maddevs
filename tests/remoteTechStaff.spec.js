import {
  mount
} from '@vue/test-utils';
import RemoteTechStaff from '@/components/Services/RemoteTechStaff';

describe('Remote tech staff', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(RemoteTechStaff);
  });

  test('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });
  
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
