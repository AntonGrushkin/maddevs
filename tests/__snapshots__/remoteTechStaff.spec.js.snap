// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remote tech staff renders correctly 1`] = `
<section
  class="remote-tech-staff main-parent-item"
  id="remote-tech-staff"
>
  <div
    class="container"
  >
    <div
      class="remote-tech-staff__content-wrapper"
    >
      <h2
        class="remote-tech-staff__main-title main-title"
      >
        Remote Tech Staff
      </h2>
       
      <div
        class="remote-tech-staff__flex-wrapper flex-wrapper"
      >
        <div
          class="remote-tech-staff-content__tech-staff teams"
        >
          <div
            class="remote-tech-staff-content__content_wrapper"
          >
            <div
              class="remote-tech-staff-content__content-wrap"
            >
              <div
                class="remote-tech-staff-content__title-wrap"
              >
                <h3
                  class="remote-tech-staff-content__title block-paragraph"
                >
                  Teams
                </h3>
                 
                <h4
                  class="remote-tech-staff-content__sub-title sub-paragraph"
                >
                  Tired of 20 or 30 people not delivering you any features?
                </h4>
              </div>
               
              <!---->
               
              <img
                alt="teams-title-symbol"
                class="remote-tech-staff-content__content-icon"
                src="@/assets/img/Services/svg/teams-symbol.svg"
              />
            </div>
             
            <p
              class="remote-tech-staff-content__paragraph paragraph"
            >
              How about only dealing with smart heads and capable hands? Enjoy our proactive rockets who will ask you to get them more work.
            </p>
          </div>
           
          <button
            class="ui-button-modal-trigger remote-tech-staff-content__ui-button-modal-trigger"
          >
            
  Get a team of ultra-fast coders

          </button>
        </div>
         
        <div
          class="remote-tech-staff-content__tech-staff individuals"
        >
          <div
            class="remote-tech-staff-content__content_wrapper"
          >
            <div
              class="remote-tech-staff-content__content-wrap"
            >
              <div
                class="remote-tech-staff-content__title-wrap"
              >
                <h3
                  class="remote-tech-staff-content__title block-paragraph"
                >
                  Individuals
                </h3>
                 
                <h4
                  class="remote-tech-staff-content__sub-title sub-paragraph"
                >
                  Seeing one, two, five guys working on a minor thing? Ha!
                </h4>
              </div>
               
              <img
                alt="individuals-title-symbol"
                class="remote-tech-staff-content__content-icon"
                src="@/assets/img/Services/svg/individuals-symbol.svg"
              />
               
              <!---->
            </div>
             
            <p
              class="remote-tech-staff-content__paragraph paragraph"
            >
              We are NOT going to bring you more people than you actually need. We'll just send one or two of our rangers to your frontier, and you'll enjoy peace of mind about IT.
            </p>
          </div>
           
          <button
            class="ui-button-modal-trigger remote-tech-staff-content__ui-button-modal-trigger"
          >
            
  Get individual proactive rockets

          </button>
        </div>
      </div>
    </div>
  </div>
</section>
`;
